packages/slider/index.tsx

```diff
@@ -0,0 +1,677 @@
+
+  state: SliderState = {
+    classList: new Set(),
+    markerStyles: [],
+  };
+
+  componentDidMount() {
+    this.initFoundation();
+  }
+
+  componentDidUpdate(prevProps: SliderProps) {
+    const {step, value, min, max, disabled} = this.props;
+    if (disabled !== prevProps.disabled) {
+      this.foundation.setDisabled(disabled!);
+    }
+    if (step !== prevProps.step) {
+      this.foundation.setStep(step!);
+    }
+    if (value !== prevProps.value) {
+      this.foundation.setValue(value!);
+    }
+    if (min !== prevProps.min) {
+      this.foundation.setMin(min!);
+    }
+    if (max !== prevProps.max) {
+      this.foundation.setMax(max!);
+    }
+  }
+
+  componentWillUnmount() {
+    if (!this.foundation) {
+      return;
+    }
+    this.foundation.destroy();
+  }
+
+  onThumbEnd?: () => void;
+
+  /**
+   * Sets the value of the slider
+   */
+  private setValue(value: number, shouldFireInput: boolean) {
+    const {discrete} = this.props;
+    this.foundation.setValue(value);
+
+    if (this.inTransit) {
+      this.onThumbEnd = () => {
+        this.setInTransit(false);
+        this.onThumbEnd = undefined;
+      };
+    }
+
+    if (shouldFireInput) {
+      this.adapter.notifyInput();
+      if (discrete) {
+        this.adapter.setMarkerValue(value);
+      }
+    }
+  }
+
+  /**
+   * Toggles the active state of the slider
+   */
+  private setActive(active: boolean) {
+    this.active = active;
+    this.toggleClass(cssClasses.ACTIVE, this.active);
+  }
+
+  /**
+   * Toggles the inTransit state of the slider
+   */
+  private setInTransit(inTransit: boolean) {
+    this.inTransit = inTransit;
+    this.toggleClass(cssClasses.IN_TRANSIT, this.inTransit);
+  }
+
+  /**
+   * Conditionally adds or removes a class based on shouldBePresent
+   */
+  private toggleClass(className: string, shouldBePresent: boolean) {
+    if (shouldBePresent) {
+      this.adapter.addClass(className);
+    } else {
+      this.adapter.removeClass(className);
+    }
+  }
+
+  /**
+   * Called when the user starts interacting with the slider
+   */
+  onDown = (downEvent: ReactMouseLikeEvent) => {
+    const {disabled} = this.props;
+    if (disabled) {
+      return;
+    }
+
+    this.preventFocusState = true;
+    this.setInTransit(!this.handlingThumbTargetEvt);
+    this.handlingThumbTargetEvt = false;
+    this.setActive(true);
+
+    const moveHandler = (moveEvent: MouseLikeEvent) => {
+      this.handleMove(moveEvent);
+    };
+
+    const moveEventType = MOVE_EVENT_MAP[downEvent.type as DownEventType];
+
+    // Note: upHandler is [de]registered on ALL potential pointer-related release event types, since some browsers
+    // do not always fire these consistently in pairs.
+    // (See https://github.com/material-components/material-components-web/issues/1192)
+    const upHandler = () => {
+      this.handleUp();
+      this.adapter.deregisterBodyInteractionHandler(moveEventType, moveHandler);
```

Comment:
This should be handled by the foundation. The general rule is that we need to implement all adapter methods (as you've done so above), and use all foundation methods. If we don't use the foundation methods, then that needs to be commented. 

Please refer to the vanilla web component to check this.